traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/models/WishlistResult.java

```diff
@@ -0,0 +1,124 @@
+    private Date fromDate;
+    private Date toDate;
+    private int total;
+    private Set<CatalogItem> items;
+
+    public WishlistResult(int skip, int take, Date fromDate, Date toDate, int total, Set<CatalogItem> items) {
+        this.skip = skip;
+        this.take = take;
+        this.fromDate = fromDate;
+        this.toDate = toDate;
+        this.total = total;
+        this.items = items;
+    }
+
+    public int getSkip() {
+        return skip;
+    }
+
+    public int getTake() {
+        return take;
+    }
+
+    public Date getFromDate() {
+        return fromDate;
+    }
+
+    public Date getToDate() {
+        return toDate;
+    }
+
+    public int getTotal() {
+        return total;
+    }
+
+    public Set<CatalogItem> getItems() {
+        return items;
+    }
+
+    public boolean isResultEquivalentTo(WishlistResult wishlistResult) {
+        return ((this.fromDate != null && this.fromDate.equals(wishlistResult.fromDate)) ||
+                (this.fromDate == null && wishlistResult.fromDate == null)) &&
+                this.toDate.equals(wishlistResult.toDate) &&
+                this.total == wishlistResult.total;
+    }
+
+    public @Nullable WishlistResult merge(WishlistResult wishlistResult) {
+        if (!this.isResultEquivalentTo(wishlistResult)) {
+            // Not mergable
+            return null;
+        }
+
+        LinkedHashSet<CatalogItem> items = new LinkedHashSet<>(this.items);
+
+        items.addAll(wishlistResult.items);
+
+        return new WishlistResult(this.skip, this.take, this.fromDate, this.toDate, this.total, this.items);
+    }
+
+    static class WishlistResultMappingFactory implements ObjectMappingFactory<WishlistResult> {
+        @Override
+        public WishlistResult instantiate(JsonReader reader) throws Exception {
+            int skip = -1;
+            int take = -1;
+            Date fromDate = null;
+            Date toDate = null;
+            int total = -1;
+            LinkedHashSet<CatalogItem> items = null;
+
+            reader.beginObject();
+            while (reader.hasNext()) {
+                String key = reader.nextName();
+
+                switch (key) {
+                    case "skip":
+                        skip = reader.nextInt();
+                        break;
+                    case "take":
+                        take = reader.nextInt();
+                        break;
+                    case "from":
+                        String fromDateString = JsonReaderHelper.nextNullableString(reader);
+                        if (fromDateString != null) {
+                            fromDate = DateHelper.parseISO8601(fromDateString);
+                        }
+                        break;
+                    case "to":
+                        String toDateString = JsonReaderHelper.nextNullableString(reader);
+                        if (toDateString != null) {
+                            toDate = DateHelper.parseISO8601(toDateString);
```

Comment:
This is not super crucial but I think `toDate` is never null

Reply:
updated

Reply:
You are still checking for null..


a way to ensure that the mapper will reject is the following:


```
toDate = DateHelper.parseISO8601(JsonReaderHelper.nextString(reader));
```

Pay attention to the methods available to you in JsonReaderHelper...or in any other tool you are using